apps/dashboard/src/components/template-store/workflow-template-modal.tsx

```diff
@@ -0,0 +1,174 @@
+import { Dialog, DialogContent, DialogFooter, DialogHeader, DialogTrigger } from '@/components/primitives/dialog';
+import { ComponentProps, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { RiArrowLeftSLine } from 'react-icons/ri';
+import { z } from 'zod';
+import { useCreateWorkflow } from '../../hooks/use-create-workflow';
+import { RouteFill } from '../icons';
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  BreadcrumbList,
+  BreadcrumbPage,
+  BreadcrumbSeparator,
+} from '../primitives/breadcrumb';
+import { Button } from '../primitives/button';
+import { CompactButton } from '../primitives/button-compact';
+import { Form } from '../primitives/form/form';
+import TruncatedText from '../truncated-text';
+import { CreateWorkflowForm } from '../workflow-editor/create-workflow-form';
+import { workflowSchema } from '../workflow-editor/schema';
+import { WorkflowCanvas } from '../workflow-editor/workflow-canvas';
+import { WorkflowResults } from './components/workflow-results';
+import { getTemplates, IWorkflowSuggestion } from './templates';
+import { WorkflowMode } from './types';
+import { WorkflowSidebar } from './workflow-sidebar';
+
+const WORKFLOW_TEMPLATES = getTemplates();
+
+export type WorkflowTemplateModalProps = ComponentProps<typeof DialogTrigger> & {
+  open?: boolean;
+  onOpenChange?: (open: boolean) => void;
+};
+
+export function WorkflowTemplateModal(props: WorkflowTemplateModalProps) {
+  const form = useForm();
+  const { submit: createFromTemplate, isLoading: isCreating } = useCreateWorkflow();
+  const [selectedCategory, setSelectedCategory] = useState<string>('popular');
+  const [suggestions, setSuggestions] = useState<IWorkflowSuggestion[]>([]);
```

Comment:
do we need these?

Reply:
Yes, going to be further used in my subsequent PR